src/datalink/mod.rs

```diff
@@ -58,6 +59,9 @@ pub enum ChannelType {
     Layer3(EtherType),
 }
 
+/// Type of timestamped Ethernet packet (Linux only)
+pub struct TimestampedEthernetPacket<'a>(Duration, EthernetPacket<'a>);
```

Comment:
This name is confusing, since this isn't actually a `#[packet]`. I think it would probably be better to just use a tuple instead.